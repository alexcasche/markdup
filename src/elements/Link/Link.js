import styled from 'styled-components'
import { Link as ReactLink } from 'react-router-dom'

const Link = styled.a`
  text-decoration: none;
  transition: all 0.15s ease-in-out;
  &:hover {
    cursor: pointer;
  }
`
const RouterLink = Link.withComponent(ReactLink)

export default Link
export { RouterLink }
