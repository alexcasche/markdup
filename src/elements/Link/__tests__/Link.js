import { BrowserRouter } from 'react-router-dom'
import Link, { RouterLink } from '../Link'

describe('Link Element', () => {
  it('renders correctly', () => {
    const tree = render(<Link to="/">Link</Link>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('RouterLink Element', () => {
  it('renders correctly', () => {
    const tree = render(
      <BrowserRouter>
        <Link to="/">Link</Link>
      </BrowserRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
