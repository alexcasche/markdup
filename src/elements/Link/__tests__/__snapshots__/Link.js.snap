// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Element renders correctly 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
}

.c0:hover {
  cursor: pointer;
}

<a
  className="c0"
  to="/"
>
  Link
</a>
`;

exports[`RouterLink Element renders correctly 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
}

.c0:hover {
  cursor: pointer;
}

<a
  className="c0"
  to="/"
>
  Link
</a>
`;
